iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestOptimisticRetry.java

```diff
@@ -74,6 +74,27 @@ public void testConcurrentOverlappingUpdates() {
 
   }
 
+  @Test
```

Comment:
Would recommend moving this to TestConflictingDataFiles.java, because phaser code in @Before and @After are not present here.

Reply:
Rest of the code LGTM

Reply:
I changed as we discussed

Reply:
Thanks!